### PR TITLE
created new condex index settings page

### PR DESCRIPTION
**Pull request by mrblippy**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] screenshots are included showing significant UI changes

##### Description of change

I'm not really sure about the wording used on this page.  This page looks fairly empty at the moment, but search boosting setting will be added to the page in a separate PR
settings1
settings2

https://github.com/openequella/openEQUELLA/issues/1337

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/components/ContentIndexSetting.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import {
+  FormControl,
+  MenuItem,
+  OutlinedInput,
+  Select
+} from "@material-ui/core";
+import { ContentIndex } from "../SearchSettingsModule";
+import * as React from "react";
+import { languageStrings } from "../../../util/langstrings";
+import { makeStyles } from "@material-ui/styles";
```
Comment (SammyIsConfused) [inline Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/components/ContentIndexSetting.tsx:10]: use @material-ui/core for makeStyles.

File: Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/ContentIndexSettings.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
+import * as React from "react";
+import {
+  templateDefaults,
+  TemplateUpdate,
+  TemplateUpdateProps
+} from "../../mainui/Template";
+import { routes } from "../../mainui/routes";
+import { Button, Card } from "@material-ui/core";
+import { makeStyles } from "@material-ui/styles";
```
Comment (SammyIsConfused) [inline Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/ContentIndexSettings.tsx:9]: Use @material-ui/core for makeStyles.

File: Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/ContentIndexSettings.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+import {
+  defaultSearchSettings,
+  getSearchSettingsFromServer,
+  saveSearchSettingsToServer,
+  SearchSettings
+} from "./SearchSettingsModule";
+import MessageInfo from "../../components/MessageInfo";
+import { Save } from "@material-ui/icons";
+import SettingsList from "../../components/SettingsList";
+import SettingsListControl from "tsrc/components/SettingsListControl";
```
[the remark this answers is not in the document]
Comment (PenghaiZhang) [reply Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/ContentIndexSettings.tsx:20]: Doesn't your Intellij help you import this ?

File: Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/ContentIndexSettings.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  }
+
+  return (
+    <>
+      <Card className={classes.spacedCards}>
+        <SettingsList>
+          <SettingsListControl
+            divider
+            primaryText={contentIndexSettingsStrings.name}
+            secondaryText={searchPageSettingsStrings.defaultSortOrderDesc}
```
Comment (PenghaiZhang) [inline Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/ContentIndexSettings.tsx:79]: `searchPageSettingsStrings` is not declared in this file or imported.  Was your IDE still happy with this?

File: Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/ContentIndexSettings.tsx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+            control={
+              <DefaultSortOrderSetting
```
Comment (PenghaiZhang) [inline Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/ContentIndexSettings.tsx:81]: Do we need `DefaultSortOrderSetting` in Content indexing page ?

File: Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/components/ContentIndexSetting.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import {
```
[the remark this answers is not in the document]
Comment (edalex-ian) [reply Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/components/ContentIndexSetting.tsx:1]: It appears to be the main control in the main `ContentIndexSettings.tsx`.

I wonder if this should actually be named `WedPageIndexSetting`. :thinking:

File: Source/Plugins/Core/com.equella.core/scalasrc/com/tle/web/settings/SettingsList.scala [hunk 7]
```diff
@@ -173,12 +173,14 @@ object SettingsList {
...
 
-  val contentIndexingSettings = CoreSettingsPage("searchpage",
-                                                 Searching,
-                                                 "setting.contentindexing.title",
-                                                 "setting.contentindexing.desc",
-                                                 "page/contentindexsettings",
-                                                 searchPrivProvider.isAuthorised)
+  val contentIndexingSettings = CoreSettingsPage(
+    "contentindexing",
+    Searching,
```
[the remark this answers is not in the document]
Comment (edalex-ian) [reply Source/Plugins/Core/com.equella.core/scalasrc/com/tle/web/settings/SettingsList.scala:178]: Yeah, be good to tidy this up @mrblippy

File: Source/Plugins/Core/com.equella.core/js/tsrc/components/SettingsListControl.tsx [hunk 8]
```diff
@@ -10,7 +10,7 @@ import { makeStyles } from "@material-ui/styles";
 interface SettingsListControlProps {
   divider?: boolean;
   primaryText: string;
-  secondaryText: string;
+  secondaryText: string | ReactNode;
```
[the remark this answers is not in the document]
Comment (PenghaiZhang) [reply Source/Plugins/Core/com.equella.core/js/tsrc/components/SettingsListControl.tsx:13]: In my PR I actually fix this by adding a question mark.

[the remark this answers is not in the document]
Comment (edalex-ian) [reply Source/Plugins/Core/com.equella.core/js/tsrc/components/SettingsListControl.tsx:13]: What issue are you trying to fix?

[the remark this answers is not in the document]
Comment (PenghaiZhang) [reply Source/Plugins/Core/com.equella.core/js/tsrc/components/SettingsListControl.tsx:13]: OK I think it's a different issue.  The issue I fixed is 'secondaryText' is not allowed to be empty.

[the remark this answers is not in the document]
Comment (mrblippy) [reply Source/Plugins/Core/com.equella.core/js/tsrc/components/SettingsListControl.tsx:13]: Yeah, i think this was something i accidentally added in while testing (can't remember what it was). It shouldn't be there. I will remove

File: Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/components/ContentIndexSetting.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (mrblippy) [reply Source/Plugins/Core/com.equella.core/js/tsrc/settings/Search/components/ContentIndexSetting.tsx:1]: Yeah, WebPageIndexSetting makes a lot more sense than ContentIndexSetting (which sounds far too generic)

File: Source/Plugins/Core/com.equella.core/scalasrc/com/tle/web/settings/SettingsList.scala (hunk 7 above)
[the remark this answers is not in the document]
Comment (mrblippy) [reply Source/Plugins/Core/com.equella.core/scalasrc/com/tle/web/settings/SettingsList.scala:178]: When i make changes to that file the format:scala task changes it back again

[the remark this answers is not in the document]
Comment (edalex-ian) [reply Source/Plugins/Core/com.equella.core/scalasrc/com/tle/web/settings/SettingsList.scala:178]: scalafmt went wacky, so ignoring for now.

**Pull request merged**